packages/core-sdk/lib/Connectors/Wallets/MetamaskConnector.ts

```diff
@@ -63,6 +67,50 @@ export class MetamaskConnector extends BaseConnector {
     return { adapter, multicall };
   }
 
+  protected async _forceNetwork({
+    chainId,
+    blockchain,
+    nativeToken,
+    publicRpc,
+  }: IBlockchainConfig) {
+    const agent = this.isAvailable() && this.getAgent();
+    if (!agent) {
+      throw new Error("Couldn't connect correctly");
+    }
+    try {
+      const success = await agent
+        .request({
+          method: "wallet_switchEthereumChain",
+          params: [{ chainId: utils.hexValue(chainId) }],
+        })
+        .then(() => true)
+        .catch(() => false);
+      if (!success) {
+        await window.ethereum.request({
+          method: "wallet_addEthereumChain",
+          params: [
+            {
+              chainId: utils.hexValue(chainId),
+              chainName: blockchain,
+              nativeCurrency: {
+                name: nativeToken.name,
+                symbol: nativeToken.symbol,
+                decimals: nativeToken.decimals,
+              },
+              rpcUrls: [publicRpc],
+            },
+          ],
+        });
+        await agent.request({
+          method: "wallet_switchEthereumChain",
+          params: [{ chainId: utils.hexValue(chainId) }],
+        });
+      }
+    } catch (e) {
+      throw new Error("Network couldn't be switch correctly.");
```

Comment:
What about creating a custom error on the errors folder with the same message but a custom constructor so we can handle it easily?